frontend/src/hooks/useLocations.js

```diff
@@ -2,15 +2,16 @@ import {useEffect, useState} from "react";
 import axios from "axios";
 
 
-export default function useLocations() {
+export default function useLocations(geoLocation) {
 
     const [locations, setLocations] = useState([])
+    const url = '/api/location' + [geoLocation? ["?lat=" + geoLocation.latitude + "&lng=" + geoLocation.longitude] : ""]
 
     useEffect(() => {
-        axios.get('/api/location')
+        axios.get(url)
             .then((response) => response.data)
             .then(setLocations)
-    }, [])
+    }, [url])
```

Comment:
Error handling! At least catch and log to prevent your app from crashing in case of a network error.